src/Adapters/Storage/Mongo/MongoTransform.js

```diff
@@ -766,19 +766,24 @@ function relativeTimeToDate(text, now = new Date()) {
 // If it is not a valid constraint but it could be a valid something
 // else, return CannotTransform.
 // inArray is whether this is an array field.
-function transformConstraint(constraint, field, count = false) {
-  const inArray = field && field.type && field.type === 'Array';
+function transformConstraint(constraint, restKey, field, count = false) {
   if (typeof constraint !== 'object' || !constraint) {
     return CannotTransform;
   }
-  const transformFunction = inArray ? transformInteriorAtom : transformTopLevelAtom;
+
+  // transformer
+  const inArray = field && field.type && field.type === 'Array';
```

Comment:
Why is checking for `&& field.type` necessary, when there is already `&& field.type === 'Array'`?

Reply:
It was already there. It was added by Florent Vilmart in this very interesting PR : [#4188](https://github.com/parse-community/parse-server/pull/4188).
But I agree. It should be changed. I also spotted other things in this file, like `key.indexOf('_p_') == 0` that should be `key.startsWith('_p_')` (no need the scan past the start of the key).
It's another PR, I think.